h2o-core/src/main/java/water/util/VecUtils.java

```diff
@@ -838,4 +837,49 @@ public void map(Chunk c, NewChunk nc) {
     }
   }
 
+  public static long seed(int cidx) {
+    return (0xe031e74f321f7e29L + ((long) cidx << 32L));
+  }
+
+
+  /**
+   * Randomly shuffle randomly a vector from a Frame
+   * */
+  
+  public static class ShuffleVecTask extends MRTask<ShuffleVecTask> {
+    @Override public void map(Chunk ic, Chunk nc) {
+      Random rng = getRNG(seed(ic.cidx()));
+      for (int i = 1; i < ic._len ; i++) {
+        int j = rng.nextInt(i); // inclusive upper bound <0,i>
```

Comment:
Why you generate the new index in rage <0, i>? Why dont you use the whole range <0, ic._len) ?

Reply:
Thats a great question. I was following the Fisher-Yates pseudocode 
for i from 1 to n do
      j ← random integer such that 0 ≤ j ≤ i

Using the entire range makes sense, I will check the difference